docs/auth.md

```diff
@@ -0,0 +1,171 @@
+# Metecho Authentication
```

Comment:
This section is much need. Thanks for writing this up!

Reply:
Oh jeez that was supposed to be in a different PR entirely. Retracting and pushing elsewhere...